### PR TITLE
Feature#25 incidence

### LINKED ISSUE #25: Incidence CRUD
# REQUIREMENTS

In the incidents appi you can only see, create and modify the data for mobile implementation and view web

The data required for this module are:
user not null
testcv not null
open not null
close null
resulttccv null
deliverytcv null
begintcv null

# DESIGN

Does not apply

# Acceptance Criteria

- [x] Appi incidence only index, create and update
- [x] CRUD web incidence
- [x] testing

### LINKED ISSUE #26: Statusymptom CRUD
# REQUIREMENTS
In the statusymptom appi you can only see, create and modify the data for mobile implementation and web view

The data required for this module are:
incidence not null
symptom not null
record not null

# DESIGN

Does not apply

# Acceptance Criteria

- [x] Appi incidence only index, create and update
- [x] CRUD web incidence
- [x] testing

### LINKED ISSUE #27: Incidence-place CRUD
# REQUIREMENTS

In the incidence-place appi you can only see, create and modify the data for mobile implementation and view web

The data required for this module are:
incidence not null
place not null
stay not null

# DESIGN

Does not apply

# Acceptance Criteria
- [x] Appi incidence-place only index, create and update
- [x] Crud web incidence place
- [x] testing

### PR DESCRIPTION
**Pull request by mariormt17**
# Description
- In this PR we implemented the functionality of the incidents on the web and api.
- In the api only the index, create and update methods were created.
  
  ## Screenshots 
Captura de Pantalla 2021-04-27 a la(s) 3 35 38 p m
Captura de Pantalla 2021-04-27 a la(s) 3 44 12 p m

### CODE REVIEW

File: alerta-covid/app/controllers/api/v1/incidences_controller.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  end
+
+  def update
+    set_variables_with_params()
+    if @incidence.update(incidence_params_update)
+      render json: @incidence, status: :ok
+    else
+      rrender json: { error: e.message }, status: :unprocessable_entity
+    end
+  end
```
Comment (MiguelOcegueraM) [inline alerta-covid/app/controllers/api/v1/incidences_controller.rb:25]: nice!

File: alerta-covid/app/views/places/_form.html.erb [hunk 2]
```diff
@@ -17,13 +17,7 @@
...
 
   <div class="field">
-    <%= form.label :qr_code %>
-    <%= form.text_field :qr_code %>
-  </div>
-
-  <div class="field">
-    <%= form.label :area_id %>
-    <%= form.text_field :area_id %>
+    <%= select_tag 'place[area_id]', options_from_collection_for_select(@areas, "id", "name", @place.area_id), include_blank: "Select Area", class: 'form-control', id: 'select-area'  %>
```
Comment (MiguelOcegueraM) [inline alerta-covid/app/views/places/_form.html.erb:20]: nice refactor

**Review by MiguelOcegueraM — DISMISSED** [review]
nice code mates, too clean

File: alerta-covid/app/models/place.rb [hunk 3]
```diff
@@ -2,8 +2,15 @@ class Place < ApplicationRecord
...
   validates :name, :area, presence: true
 
-  def self.generate_qr_code(place_id)
-    qr_code = RQRCode::QRCode.new(place_id)
+  has_many :incidence_places
+  has_many :incidences, through: :incidence_places
+
+
+  after_create :generate_qr_code
+  after_destroy :destroy_qr_code
```
Comment (ArielMD) [inline alerta-covid/app/models/place.rb:10]: Nice, callbacks

**Review by ArielMD — DISMISSED** [review]
Nice job team

**Review by FranckZnt19 — CHANGES_REQUESTED** [review]
all it's fine, but this task does not have unit tests neither in web module nor in appi. it's  necessary to place them.

File: alerta-covid/app/models/covid_test.rb [hunk 4]
```diff
@@ -1,3 +1,4 @@
 class CovidTest < ApplicationRecord
   validates :name, presence: true
+  has_many  :incidence
```
Comment (jaimelr) [inline alerta-covid/app/models/covid_test.rb]: I would expect this to be plural: `incidences`, what do you think? 
> The name of the other model is pluralized when declaring a has_many association.

See: Rails Guides [2.3 The has_many association](https://guides.rubyonrails.org/association_basics.html#has-many-association-reference)

File: alerta-covid/app/models/user.rb [hunk 5]
```diff
@@ -5,6 +5,7 @@ class User < ApplicationRecord
          :recoverable, :rememberable, :validatable, :confirmable
 
   belongs_to :role
+  has_many :incidence
```
Comment (jaimelr) [inline alerta-covid/app/models/user.rb]: See my previous comment.

File: alerta-covid/app/controllers/incidences_controller.rb [hunk 6]
```diff
@@ -17,13 +18,16 @@ def new
   end
 
   # GET /incidences/1/edit
-  def edit; end
+  def edit
+    @user_symptoms = @incidence.symptoms.collect{|p|p.id}
```
Comment (jaimelr) [inline alerta-covid/app/controllers/incidences_controller.rb]: `@user_symptoms = @incidence.symptoms.collect { |p| p.id }`

File: alerta-covid/app/controllers/incidences_controller.rb [hunk 7]
```diff
@@ -55,15 +56,27 @@ def set_incidence
...
   end
 
   def set_variables
     @tests = CovidTest.all.map { |test| [test.name, test.id] }
     @symptoms = Symptom.all
+    @places = Place.all
+  end
+
+  def set_variables_with_params
+    puts("desde set params #{params.require(:incidence).inspect}")
```
Comment (jaimelr) [inline alerta-covid/app/controllers/incidences_controller.rb]: Is this a debug line?

**Review by jaimelr — COMMENTED** [review]
Please review the indentation of the files (must be two spaces), check on the spaces of assignments for example: `@areas=Area.all`

Also this PR seems to include changes that are not related to Incidences, usually Pull Request must focus on one specific topic is possible. I'm seeing Incidences, Department and QR code updates in the same branch. If they all need to be in this PR, please add it to the PR description.

**Review by FranckZnt19 — APPROVED** [review]

**Review by ArielMD — APPROVED** [review]
Nice

**Review by jaimelr — APPROVED** [review]

**Pull request merged**